test/Cli/Func.E2E.Tests/README.md

```diff
@@ -0,0 +1,230 @@
+var result = funcStartCommand
+    .WithWorkingDirectory(WorkingDirectory)
+    .Execute(commandArgs.ToArray());
+```
+
+### 9. Validate the Results
+
+Validate the command output contains the expected results based on test parameters:
+
+```csharp
+if (someCondition)
+{
+    result.Should().HaveStdOutContaining("expected output 1");
+}
+else
+{
+    result.Should().HaveStdOutContaining("expected output 2");
+}
+```
+
+You may also create a custom condition and then call the method like so:
+
+```csharp
+// Validate inproc6 host was started
+result.Should().StartInProc6Host();
+
+```
+
+## Testing with Fixtures
+
+The steps are similar for creating a `func start` test with a fixture, except you can skip the setup logic that calls `func init` and `func new` as the fixture handles this for you. Please ensure that the test that are being added to the fixture DO NOT change the environment or add any extra variables or config, as that may cause problems with the existing tests.
+
+## Complete Example
+
+```csharp
+public async Task Start_DotnetIsolated_Test_EnableAuthFeature(
+    string authLevel,
+    bool enableAuth,
+    string expectedResult)
+{
+    int port = ProcessHelper.GetAvailablePort();
+    string methodName = "Start_DotnetIsolated_Test_EnableAuthFeature";
+    string uniqueTestName = $"{methodName}_{authLevel}_{enableAuth}";
+    
+    // Setup the function app
+    await FuncInitWithRetryAsync(uniqueTestName, new[] { ".", "--worker-runtime", "dotnet-isolated" });
+    await FuncNewWithRetryAsync(uniqueTestName, new[] { ".", "--template", "HttpTrigger", "--name", "HttpTrigger", "--authlevel", authLevel });
+    
+    // Create and configure the start command
+    var funcStartCommand = new FuncStartCommand(FuncPath, methodName, Log);
+    funcStartCommand.ProcessStartedHandler = async (process) =>
+    {
+        await ProcessHelper.ProcessStartedHandlerHelper(port, process, funcStartCommand.FileWriter, "HttpTrigger");
+    };
+    
+    // Build command arguments
+    var commandArgs = new List<string> { "start", "--verbose", "--port", port.ToString() };
+    if (enableAuth)
+    {
+        commandArgs.Add("--enableAuth");
+    }
+    
+    // Execute and validate
+    var result = funcStartCommand
+        .WithWorkingDirectory(WorkingDirectory)
+        .Execute(commandArgs.ToArray());
+    
+    if (string.IsNullOrEmpty(expectedResult))
+    {
+        result.Should().HaveStdOutContaining("\"status\": \"401\"");
+    }
+    else
+    {
+        result.Should().HaveStdOutContaining("Selected out-of-process host.");
+    }
+}
+```
+
+## How to Build and Run Tests Locally
+
+## How to Build and Run Tests Locally
+
+1. Run `dotnet build` on `Azure.Functions.Cli.E2E.Tests.csproj`.
+
+2. Navigate to `CORE_TOOLS_REPO_ROOT\out\bin\Azure.Functions.Cli.E2E.Tests\debug` and add a `templates` file if it doesn't already exist. Copy the existing templates file from the core tools installation on your local machine to the debug directory.
```

Comment:
Is there another way to get templates without copying them from an exisiting download?

I do have a script I will eventually check in and can mention here too